src/eventstore/EventStoreSubscriber.cs

```diff
@@ -5,125 +5,72 @@
-        private long? _startingPosition;
-        private int _maxLiveQueueSize;
-        private object _subscription;
-        private int _catchupPageSize;
+        private EventStoreClient _eventStoreClient;
+        private ulong? _startingPosition;
+        private StreamSubscription _subscription;
         private string _streamName;
         private bool _liveOnly;
-
-        private long _lastNonLiveEventNumber = long.MinValue;
-        private long? _lastReceivedEventNumber;
+        private bool _isSubscribed;
+        private ulong? _lastProcessedEventNumber;
         private int _eventsProcessed;
-        private bool _catchingUp = true;
-
-        private TimeSpan _heartbeatTimeout;
-        private DateTime _lastHeartbeat;
-        private Timer _heartbeatTimer;
-        private bool _usingHeartbeats;
+        private ulong _liveEventThreshold;
+        private ulong _lastStreamPosition;
+        private DateTime _lastStreamPositionTimestamp;
 
-        private BlockingCollection<ResolvedEvent> _queue;
         private IDispatcher<ResolvedEvent> _dispatcher;
-        private long _lastDispatchedEventNumber;
-
         private ILogger _logger;
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="EventStoreSubscriber"/> class.
-        /// </summary>
-        /// <param name="connection">Eventstore connection.</param>
-        /// <param name="dispatcher">Dispatcher.</param>
-        /// <param name="streamName">Stream name to push events into.</param>
-        /// <param name="logger">Logger.</param>
-        /// <param name="startingPosition">Starting Position.</param>
-        /// <param name="catchUpPageSize">Catchup page size.</param>
-        /// <param name="upperQueueBound">Upper Queue Bound.</param>
-        /// <param name="heartbeatFrequency">Frequency of heartbeat.</param>
-        /// <param name="heartbeatTimeout">Timeout of heartbeat.</param>
-        /// <param name="maxLiveQueueSize">Maximum size of the live queue.</param>
-        [Obsolete("Please use new static method CreateCatchUpSubscirptionFromPosition, this constructor will be removed in the future.")]
-        public EventStoreSubscriber(
-            IEventStoreConnection connection,
+        private EventStoreSubscriber(
+            EventStoreClient eventStoreClient,
             IDispatcher<ResolvedEvent> dispatcher,
             string streamName,
             ILogger logger,
-            int? startingPosition,
-            int catchUpPageSize = 1024,
-            int upperQueueBound = 2048,
-            TimeSpan? heartbeatFrequency = null,
-            TimeSpan? heartbeatTimeout = null,
-            int maxLiveQueueSize = 10000)
-            => Init(connection, dispatcher, streamName, logger, heartbeatFrequency, heartbeatTimeout, startingPosition, catchUpPageSize, upperQueueBound, maxLiveQueueSize);
+            ulong? startingPosition,
+            ulong liveEventThreshold)
```

Comment:
Similar to above, if we only support up to longs is there any point in this being a ulong?